docs/api/create_docker_container.md

```diff
@@ -0,0 +1,78 @@
+
+## Supported commands
+
+| Builder method                          | Description                                                                                                     |
+|-----------------------------------------|-----------------------------------------------------------------------------------------------------------------|
+| `WithDockerEndpoint`                    | Sets the Docker daemon socket to connect to.                                                                    |
+| `WithAutoRemove`                        | Will remove the stopped container automatically like `--rm`.                                                    |
+| `WithCleanUp`                           | Will remove the container automatically after all tests have been run.                                          |
+| `WithLabel`                             | Applies metadata to the container e.g. `-l, --label testcontainers=awesome`.                                    |
+| `WithResourceReaperSessionId`           | Assigns a Resource Reaper session id to the container.                                                          |
+| `WithRegistryAuthentication`            | Sets the basic authentication against a private Docker registry.                                                |
+| `WithImage`                             | Specifies an image to derive the container from.                                                                |
+| `WithImagePullPolicy`                   | Specifies an image pull policy to determine when an image is pulled e.g. `--pull "always"\|"missing"\|"never"`. |
+| `WithName`                              | Sets the container name e.g. `--name testcontainers`.                                                           |
+| `WithHostname`                          | Sets the container hostname e.g. `--hostname testcontainers`.                                                   |
+| `WithMacAddress`                        | Sets the container MAC address e.g. `--mac-address "00:80:41:ae:fd:7e"`.                                        |
+| `WithWorkingDirectory`                  | Specifies or overrides the `WORKDIR` for the instruction sets.                                                  |
+| `WithEntrypoint`                        | Specifies or overrides the `ENTRYPOINT` that runs the executable.                                               |
+| `WithCommand`                           | Specifies or overrides the `COMMAND` instruction provided in the Dockerfile.                                    |
+| `WithEnvironment`                       | Sets an environment variable in the container e.g. `-e, --env "MAGIC_NUMBER=42"`.                               |
+| `WithExposedPort`                       | Exposes a port inside the container e.g. `--expose=80`.                                                         |
```

Comment:
This actually does not do anything, it is just a form of documenting exposed ports on the container. The ports are accessible as well if you don't use the flag. Same purpose as `EXPOSE` in Dockerfile.

Reply:
If the port is not exposed via `Dockerfile` you can expose it via `WithExposedPort`.

Reply:
Oh you are probably right, `-p` will implicitly expose a port in the CLI I assume. In the case of tc-java, we have slightly different semantics for expose. `container.withExposedPort(80)` will be the same as doing a port publishing like `-p 80`.